js/renovation/widget.tsx

```diff
@@ -157,56 +167,62 @@ export default class Widget extends JSXComponent(WidgetProps) {
           showTimeout: _feedbackShowTimeout,
         });
 
-      return () => active.off(this.widgetRef, { selector, namespace });
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      return (): any => active.off(this.widgetRef, { selector, namespace });
     }
 
     return undefined;
   }
 
   @Effect()
-  clickEffect() {
-    const { name, onClick } = this.props;
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  clickEffect(): any {
+    const { name, onClick, disabled } = this.props;
     const namespace = name;
 
-    if (onClick) {
+    if (onClick && !disabled) {
       dxClick.on(this.widgetRef,
         (e) => onClick(e),
         { namespace });
-
-      return () => dxClick.off(this.widgetRef, { namespace });
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      return (): any => dxClick.off(this.widgetRef, { namespace });
     }
 
     return undefined;
   }
 
   @Method()
-  focus() {
+  focus(): void {
     focus.trigger(this.widgetRef);
   }
 
   @Effect()
-  focusEffect() {
-    const { disabled, focusStateEnabled, name } = this.props;
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  focusEffect(): any {
+    const {
+      disabled, focusStateEnabled, name, onFocusin, onFocusout,
+    } = this.props;
     const namespace = `${name}Focus`;
     const isFocusable = focusStateEnabled && !disabled;
 
     if (isFocusable) {
       focus.on(this.widgetRef,
-        (e) => { !e.isDefaultPrevented() && (this.focused = true); },
-        (e) => { !e.isDefaultPrevented() && (this.focused = false); },
+        (e) => { onFocusin?.(e); !e.isDefaultPrevented() && (this.focused = true); },
+        (e) => { onFocusout?.(e); !e.isDefaultPrevented() && (this.focused = false); },
```

Comment:
Should we call `onFocusIn` and `onFocusOut` if `e.isDefaultPrevented() === true`?

Reply:
I think, we shouldn't. Fixed, thx